sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -133,6 +132,38 @@ object ExtractEquiJoinKeys extends Logging with PredicateHelper {
 }
 
 /**
+  * A pattern that collects the filter and inner joins.
```

Comment:
Is it much more work to extract all the filters? For example if there is a filter after the inner join of input and plan 1